udata/harvest/forms.py

```diff
@@ -11,6 +11,38 @@
 __all__ = 'HarvestSourceForm', 'HarvestSourceValidationForm'
 
 
+class HarvestConfigField(fields.DictField):
+    '''
+    A DictField with extras validations on known configurations
+    '''
+    def get_backend(self, form):
+        return [
+            b for b in list_backends() if b.name == form.backend.data
+        ][0]
+
+    def get_specs(self, backend, key):
+        candidates = [f for f in backend.filters if f.key == key]
+        return candidates[0] if len(candidates) == 1 else None
+
+    def pre_validate(self, form):
+        if self.data:
+            backend = self.get_backend(form)
+            # Validate filters
+            for f in (self.data.get('filters', None) or []):
```

Comment:
No need for `, None` I think.

Reply:
Or better `self.data.get('filters', [])`

Reply:
Sadly `self.data.get('filters', [])` doesn't cover the case where `filters is None` or I have to write `self.data.get('filters', []) or []`.
I'll go for `self.data.get('filters') or []`